modules/afsocket/afinet-dest.c

```diff
@@ -118,6 +118,43 @@ afinet_dd_set_tls_context(LogDriver *s, TLSContext *tls_context)
                                         afinet_dd_verify_callback, self);
 }
 
+void
+afinet_dd_add_failovers(LogDriver *s, GList *failovers)
+{
+  AFInetDestDriver *self = (AFInetDestDriver *)s;
+  if (!self->server_candidates)
+    {
+      self->server_candidates = g_list_append(self->server_candidates, self->hostname);
+    }
+  while (failovers)
+    {
+      self->server_candidates = g_list_append(self->server_candidates, g_strdup(failovers->data));
+      g_free(failovers->data);
```

Comment:
If g_list_concat is not appropriate, I would still remove this g_strdup g_free combination. We could simply pass the reference to the other list.

Reply:
the string_list rule in the grammar generates a GList with the data member pointing to a g_strdup()-ed string. So concatentating the two lists should just work.